polkadot/api/src/lib.rs

```diff
@@ -135,12 +136,21 @@ pub trait PolkadotApi {
 	/// Get the index of an account at a block.
 	fn index(&self, at: &Self::CheckedBlockId, account: AccountId) -> Result<Index>;
 
+	/// Get the active parachains at a block.
+	fn active_parachains(&self, at: &Self::CheckedBlockId) -> Result<Vec<ParaId>>;
+
+	/// Get the validation code of a parachain at a block. If the parachain is active, this will always return `Some`.
+	fn parachain_code(&self, at: &Self::CheckedBlockId, parachain: ParaId) -> Result<Option<Vec<u8>>>;
```

Comment:
IMO type-safety is only useful when we have guarded types (for example, that this type can only be instantiated with valid WASM that fulfills the parachain ABI). But since any arbitrary bytes can be registered as the parachain code, what do we actually gain from writing a wrapper?

Reply:
Just the fact that the arbitrary bytes type hanging around in the code is actually a parachain code, not anything else.